pkg/types/ccipocr3/generic_types.go

```diff
@@ -62,6 +62,27 @@ func (s *SeqNumRange) SetEnd(v SeqNum) {
 	s[1] = v
 }
 
+// Limit returns a range limited up to n elements by truncating the end if necessary.
+// Example: [1 -> 10].Limit(5) => [1 -> 5]
+func (s *SeqNumRange) Limit(n uint64) SeqNumRange {
+	limitedRange := NewSeqNumRange(s.Start(), s.End())
+
+	numElems := s.End() - s.Start() + 1
```

Comment:
nit: I wonder start > end. It might add an extra safety to return an empty range. But not sure if this case is possible